detector/rules.go

```diff
@@ -20,46 +20,43 @@ package detector
 import (
 	"encoding/json"
 	"fmt"
-	"io"
 	"io/ioutil"
 	"os"
 
-	"github.com/markbates/pkger"
+	"go.elastic.co/go-licence-detector/assets"
 )
 
-const embeddedRulesFile = "go.elastic.co/go-licence-detector:/assets/rules.json"
-
 // rulesFile represents the structure of the rules file.
 type rulesFile struct {
-	Allowlist  []string `json:"allowlist"`
+	Allowlist []string `json:"allowlist"`
 	Maybelist []string `json:"maybelist"`
 }
 
 // Rules holds rules for the detector.
 type Rules struct {
-	AllowList  map[string]struct{}
+	AllowList map[string]struct{}
 	Maybelist map[string]struct{}
 }
 
 // LoadRules loads rules from the given path. Embedded rules file is loaded if the path is empty.
 func LoadRules(path string) (*Rules, error) {
-	var f io.ReadCloser
-	var err error
+	var ruleBytes []byte
 
 	if path == "" {
-		f, err = pkger.Open(embeddedRulesFile)
+		ruleBytes = assets.Rules
 	} else {
-		f, err = os.Open(path)
-	}
+		f, err := os.Open(path)
 
-	if err != nil {
-		return nil, fmt.Errorf("failed to open rules file: %w", err)
-	}
-	defer f.Close()
+		if err != nil {
+			return nil, fmt.Errorf("failed to open rules file: %w", err)
+		}
+		defer f.Close()
+
+		ruleBytes, err = ioutil.ReadAll(f)
```

Comment:
`ioutil` is deprecated

```suggestion
		ruleBytes, err = io.ReadAll(f)
```